TeacherWorkout.Domain/Lessons/GetLessons.cs

```diff
@@ -0,0 +1,33 @@
+using System.Collections.Generic;
+using TeacherWorkout.Domain.Common;
+using TeacherWorkout.Domain.Models;
+using TeacherWorkout.Domain.Models.Inputs;
+
+namespace TeacherWorkout.Domain.Lessons
+{
+    public class GetLessons : IOperation<GetLessons.Input, PaginatedResult<Lesson>>
+    {
+        public class Input : PaginationInput
+        {
+            public int? ThemeId { get; set; }
+        }
+        
+        private readonly ILessonRepository _repository;
+
+        public GetLessons(ILessonRepository repository)
+        {
+            _repository = repository;
+        }
+        
+        public PaginatedResult<Lesson> Execute(Input input)
+        {
+            var filterList = new List<IFilter>();
+            if (input.ThemeId.HasValue)
+            {
+                filterList.Add(new ThemeFilter(input.ThemeId.Value));
+            }
```

Comment:
Won't it be cleaner to have a complex graph type together with a complex model used to filter lessons? By following this approach we can also add more filters without modifying the query contract.